projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -1,24 +1,590 @@
+                .setStopLon(Double.parseDouble(record.get("stop_lon")))
+                .setLocationType(Integer.parseInt(record.get("location_type")))
+                .setParentStation(record.get("parent_station"))
+                .setZoneId(Integer.parseInt(record.get("zone_id")))
+                .build()
+                .writeDelimitedTo(output);
+      }
+      output.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void saveTranslations(Path csvPath, String path) {
+     
+      // where we serialize the messages
+      try(FileOutputStream output = new FileOutputStream(path);)
+
+    {  CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT
+      .withHeader(
+        "trans_id",
+        "lang",
+        "translation"));
+      // Creat iterator - to create from each record a message
+      Iterator<CSVRecord> iterator = parser.iterator();
+      // We need to ignore the first CSVRecord which holds the csv headers
+      iterator.next();
+      while (iterator.hasNext()) {
+        CSVRecord record = iterator.next();
+        Translation.Builder translation = Translation.newBuilder();
+            translation.setTransId(record.get("trans_id"))
+            .setLang(record.get("lang"))
+            .setTranslation(record.get("translation"))
+            .build()
+            .writeDelimitedTo(output);
+      }
+      output.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void saveTrips(Path csvPath, String path) {
+     
+      // where we serialize the messages
+      try(FileOutputStream output = new FileOutputStream(path);)
+
+    {  CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT
+      .withHeader(
+        "route_id",
+        "service_id",
+        "trip_id",
+        "trip_headsign",
+        "direction_id",
+        "shape_id"));
+      // Creat iterator - to create from each record a message
+      Iterator<CSVRecord> iterator = parser.iterator();
+      // We need to ignore the first CSVRecord which holds the csv headers
+      iterator.next();
+      while (iterator.hasNext()) {
+        CSVRecord record = iterator.next();
+        Trip.Builder trip = Trip.newBuilder();
+        trip.setRouteId(Integer.parseInt(record.get("route_id")))
+            .setServiceId(Integer.parseInt(record.get("service_id")))
+            .setTripId(record.get("trip_id"))
+            .setTripHeadsign(record.get("trip_headsign"))
+            .setDirectionId(Integer.parseInt(record.get("direction_id")))
+            .setShapeId(record.get("shape_id"))
+            .build()
+            .writeDelimitedTo(output);
+      }
+      output.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  //TEST: test by deserialize and reach last message and print it

```

Comment:
It's great that you added a test, but let's test it in a real Java test.
I suggest you remove it and we add it to the next PR.

Reply:
Correct!